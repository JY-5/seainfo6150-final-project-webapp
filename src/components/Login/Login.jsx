import React from 'react';
import { useState } from 'react';
import styles from './Login.module.css';
import { withRouter } from 'react-router';

const Login = (props) => {
    const [username, setUsername] = useState('');
    const [isDisabled, setIsDisabled] = useState(true);
    const [isPending, setIsPending] = useState(false);

    const onChange = (e) => {
        setUsername(e.target.value);
        setIsDisabled(!e.target.value);
    };

    const login = () => {
        props.history.goBack();
        setIsPending(true);
        setIsPending(false);
        props.changeLogIn(username);
    };

    return (
        <div className={styles.login}>
            <div className={styles.input}>
                <label className={styles.usernameLabel}>Username: 
                    <input className={styles.username} disabled={isPending} onChange={onChange} type="text"/>
                </label>
                <label className={styles.passwordLabel}>Password: 
                    <input className={styles.password} disabled={isPending} type="password"/>
                </label>   
                <button className={styles.loginButton} onClick={login} disabled={isDisabled || isPending} >{ isPending ? "..." : "Login"}</button>
            </div>             
        </div>
    )
}

export default withRouter(Login)