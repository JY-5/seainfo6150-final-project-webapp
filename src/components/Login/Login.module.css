.login {
    display: flex;
    align-items: center; 
    justify-content: center;
    height: 500px;
    font-family: 'Prompt', sans-serif;
}

.input {
    display: flex;
    flex-direction: column;
}

.username, .password {
    font-size: 1rem;
    margin-left: 0.8rem;
}

.usernameLabel {
    margin-bottom: 2rem;
}

.passwordLabel {
    margin-bottom: 2rem;
}

.loginButton {
    flex: 1 100%;
    background-color: #74b840;
    color: white;
    border: none;
    height: 3rem;
    font-size: 1rem;
    font-weight: 600;
    border-radius: 5px;
    padding: 0.4rem;
    text-decoration: none;
    text-align: center;
}