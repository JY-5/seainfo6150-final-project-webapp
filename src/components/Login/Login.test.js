import React from "react";
import Login from './Login';
import { render } from "@testing-library/react";
import {BrowserRouter as Router} from "react-router-dom";

describe("Add Login tests", () => {
    it("renders correctly", () => {
      const { container } = render(<Router><Login/></Router>);
      expect(container).toMatchSnapshot();
    });
});