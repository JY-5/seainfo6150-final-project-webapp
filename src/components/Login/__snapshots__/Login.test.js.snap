// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Login tests renders correctly 1`] = `
<div>
  <div
    class="login"
  >
    <div
      class="input"
    >
      <label
        class="usernameLabel"
      >
        Username:
        <input
          class="username"
          type="text"
        />
      </label>
      <label
        class="passwordLabel"
      >
        Password:
        <input
          class="password"
          type="password"
        />
      </label>
      <button
        class="loginButton"
        disabled=""
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
